Objects/interpolationobject.c

```diff
@@ -0,0 +1,217 @@
+/* t-string Interpolation object implementation */
+
+#include "Python.h"
+#include "pycore_initconfig.h"      // _PyStatus_OK
+#include "pycore_interpolation.h"
+
+static int
+_conversion_converter(PyObject *arg, PyObject **conversion)
+{
+    if (arg == Py_None) {
+        return 1;
+    }
+
+    if (!PyUnicode_Check(arg)) {
+        PyErr_Format(PyExc_TypeError,
+            "%.200s() %.200s must be %.50s, not %.50s",
+            "Interpolation", "argument 'conversion'", "str",
+            arg == Py_None ? "None" : Py_TYPE(arg)->tp_name);
```

Comment:
Is there a need to use `%.200s()` for `Interpolation()` and `argument 'conversion'`? In addition, `arg == Py_None` will never be true as you're early returning. Also, instead of `Py_TYPE(arg)->tp_name`, you could use `%T`.

Reply:
Done in [`5782897` (#132662)](https://github.com/python/cpython/pull/132662/commits/5782897688200183b052339bba63767bafe161a0)